### ISSUE TITLE
QMI Signals should be more Qt-like

### ISSUE THREAD

**Issue by jorisvr** [teamdiamond/qmi#58]
* Subscribe to signal via the RPC proxy of its object, instead of just specifying "context_name", "object_name", "signal_name".
* Instead of _registering_ signal types at run-time, write the signal declarations as class members, similar to `pyqtSignal`.
* Such signal types should then carry over into RPC proxies as a type of object that can be used to subscribe to.

For example, an RPC class can declare a signal type as follows:
```
class MyClass(QMI_RpcObject):
    sigHello = qmiSignal([str, int])
    ....
```

Another task can subscribe to this signal as follows:
```
    proxy = qmi.get_rpc_object_by_name("context_name.my_object_name")
    proxy.sigHello.subscribe(my_receiver)
```

This is a pretty big change in the QMI publish/subscribe mechanism.

**Comment by jorisvr** [comment]
This is now implemented (commit 1783739fb6b2a5019b4968f6b3f32561f172ca43).
Pushed to a temporary branch signals_cleanup, to be merged into master as soon as the qinc-demonstrator code has been updated to match the new API.

Note: In Qt, when an object is removed, all subscriptions to its signals are automatically removed as well. We have not implemented this in QMI. Subscriptions on a removed object remain in place, and will become active if a new object is created with the same name. It would be nicer to automatically remove such subscriptions, but this is difficult to do correctly in case of remote subscribers due to concurrency issues. We can leave this for now and fix if it ever becomes important.

**Comment by jorisvr** [comment]
Done (commit d6efb37f7711ff7774e5762becf67ddaa9afe4dd).

New issue #59 created for the handling of subscriptions on signals of removed objects.
